azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchImpl.java

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.executor.container;
+
+import azkaban.Constants;
+import azkaban.Constants.ContainerizedExecutionManagerProperties;
+import azkaban.DispatchMethod;
+import azkaban.executor.AbstractExecutorManagerAdapter;
+import azkaban.executor.ExecutableFlow;
+import azkaban.executor.Executor;
+import azkaban.executor.ExecutorApiGateway;
+import azkaban.executor.ExecutorLoader;
+import azkaban.executor.ExecutorManagerException;
+import azkaban.executor.Status;
+import azkaban.metrics.CommonMetrics;
+import azkaban.utils.FileIOUtils.LogData;
+import azkaban.utils.Pair;
+import azkaban.utils.Props;
+import com.google.common.util.concurrent.RateLimiter;
+import java.io.IOException;
+import java.lang.Thread.State;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class is used to dispatch execution on Containerized infrastructure. Each execution will
+ * be dispatched in single container. This way, it will bring isolation between all the
+ * executions.
+ */
+@Singleton
+public class ContainerizedDispatchImpl extends AbstractExecutorManagerAdapter {
```

Comment:
I think the name can be ContainerizedDispatch itself, no need to keep the "impl" suffix.